src/App.js

```diff
@@ -74,6 +75,11 @@ class App extends Component {
   constructor(props){
     super(props);
 
+    const welcomeShown = JSON.parse(localStorage.getItem("welcomeShown"));
+    if (!welcomeShown) {
```

Comment:
at this point the welcome has not been shown yet. Even if it works is not intuitive. It shoud be set after the welcome dialog has been shown.